### PR TITLE
add: stardust-flatland

### PR DESCRIPTION
**Pull request by Owen-sz**

### CODE REVIEW

File: anda/stardust/flatland/stardust-flatland.spec [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+%description
+Flatland for StardustXR.
+
+%prep
+%autosetup -n flatland-%commit
+%cargo_prep_online
+
+%build
+%cargo_build
```
Comment (madonuko) [inline anda/stardust/flatland/stardust-flatland.spec]:
```suggestion
```
Don't build twice

File: anda/stardust/flatland/stardust-flatland.spec [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+# Exclude input files from mangling
+%global __brp_mangle_shebangs_exclude_from ^/usr/src/.*$
+
+Name:           stardust-flatland
+Version:        %commit_date.%shortcommit
+Release:        1%?dist
+Summary:        Flatland for StardustXR.
+URL:            https://github.com/StardustXR/flatland
+Source0:        https://github.com/StardustXR/flatland/archive/%commit/flatland-%commit.tar.gz
```
Comment (madonuko) [inline anda/stardust/flatland/stardust-flatland.spec]:
```suggestion
Source0:        %url/archive/%commit/flatland-%commit.tar.gz
```

File: anda/stardust/flatland/update.rhai [hunk 3]
```diff
@@ -0,0 +1,7 @@
+if filters.contains("nightly") {
```
Comment (madonuko) [inline anda/stardust/flatland/update.rhai]: You don't need this if statement.

File: anda/stardust/flatland/stardust-flatland.spec [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+License:        MIT
+BuildRequires:  cargo cmake anda-srpm-macros cargo-rpm-macros mold
+Requires:       libgcc glibc
```
Comment (madonuko) [inline anda/stardust/flatland/stardust-flatland.spec]: You probably don't need this

File: anda/stardust/flatland/stardust-flatland.spec [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+Provides:       flatland
+
+%description
+Flatland for StardustXR.
```
Comment (madonuko) [inline anda/stardust/flatland/stardust-flatland.spec]:
```suggestion
%summary
```

Comment (madonuko) [reply anda/stardust/flatland/stardust-flatland.spec]: Also, `Summary:` usually doesn't end with `.`s

**Review by madonuko — CHANGES_REQUESTED** [review]

File: anda/stardust/flatland/stardust-flatland.spec [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+Name:           stardust-flatland
+Version:        %commit_date.%shortcommit
+Release:        1%?dist
+Summary:        Flatland for StardustXR.
+URL:            https://github.com/StardustXR/flatland
+Source0:        %url/archive/%commit/flatland-%commit.tar.gz
+License:        MIT
+BuildRequires:  cargo cmake anda-srpm-macros cargo-rpm-macros mold
+
+Provides:       flatland
```
Comment (madonuko) [inline anda/stardust/flatland/stardust-flatland.spec:16]: Packager

File: anda/stardust/flatland/update.rhai (hunk 3 above)
Comment (madonuko) [reply anda/stardust/flatland/update.rhai]: remove the `if ...` and the brackets

**Review by madonuko — CHANGES_REQUESTED** [review]

File: anda/stardust/flatland/anda.hcl [hunk 7]
```diff
@@ -0,0 +1,8 @@
+project pkg {
+	rpm {
+		spec = "stardust-flatland.spec"
+	}
+labels {
```
Comment (madonuko) [inline anda/stardust/flatland/anda.hcl]: indentation

File: anda/stardust/flatland/update.rhai [hunk 8]
```diff
@@ -0,0 +1 @@
+rpm.global("commit", gh_commit("StardustXR/flatland"));
```
Comment (madonuko) [inline anda/stardust/flatland/update.rhai:1]:
```suggestion
rpm.global("commit", gh_commit("StardustXR/flatland"));
if rpm.changed() {
  rpm.release();
  rpm.global("commit_date", date());
}
```

**Review by madonuko — CHANGES_REQUESTED** [review]

**Review by madonuko — DISMISSED** [review]

**Review by korewaChino — DISMISSED** [review]
it do build

**Review by madonuko — DISMISSED** [review]

**Review by madonuko — DISMISSED** [review]

**Review by madonuko — APPROVED** [review]

**Comment by raboneko** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|f39|[](https://github.com/terrapkg/packages/pull/2095)|
|✅|f40|[](https://github.com/terrapkg/packages/pull/2096)|
|✅|f41|[](https://github.com/terrapkg/packages/pull/2097)|
|✅|el9|[](https://github.com/terrapkg/packages/pull/2098)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/10943072340) for details

**Pull request merged**